src/win/win-helpers.h

```diff
@@ -28,9 +29,79 @@ namespace librealsense
 
         std::vector<std::string> tokenize(std::string string, char separator);
 
+        std::wstring instance_id_from_device_path( LPCWSTR path );
         bool parse_usb_path_multiple_interface(uint16_t & vid, uint16_t & pid, uint16_t & mi, std::string & unique_id, const std::string & path, std::string & device_guid);
         bool parse_usb_path_single_interface(uint16_t & vid, uint16_t & pid, std::string & serial, const std::string & path);
         bool get_usb_descriptors(uint16_t device_vid, uint16_t device_pid, const std::string& device_uid, std::string& location, usb_spec& spec, std::string& serial);
+        bool get_usb_device_descriptors( DEVINST devinst, uint16_t device_vid, uint16_t device_pid, const std::string& device_uid, std::string& location, usb_spec& spec, std::string& serial, std::string& composite_uid );
+
+        /*
+            Configuration Management (CM) tree node.
+            See https://docs.microsoft.com/en-gb/windows/win32/api/cfgmgr32/
+
+            A basic wrapper around DEVINST and the functionality we need around it, for ease of use.
+        */
+        class cm_node
+        {
+            DEVINST _devinst;
+
+        public:
+            cm_node() : _devinst( 0 ) {}
+            cm_node( DEVINST devinst ) : _devinst( devinst ) {}
+
+            static cm_node from_instance_id( std::wstring const& instance_id );
+            static cm_node from_device_path( LPCWSTR device_path )
+            {
+                std::wstring instance_id = instance_id_from_device_path( device_path );
+                return from_instance_id( instance_id );
+            }
+            static cm_node root();
+
+            DEVINST get() const { return _devinst; }
+            bool valid() const { return get() != 0; }
+
+            operator DEVINST() const { return _devinst; }
+
+            operator DEVINST* () { return &_devinst; }
+            operator DEVINST const* () const { return &_devinst; }
+
+            std::string get_id() const;
+            std::string get_uid() const;
+
+            cm_node get_parent() const;
+            cm_node get_sibling() const;
+            cm_node get_child() const;
+
+            std::string get_property( DEVPROPKEY const& property ) const;
+
+            /*
+                Iterate through all nodes (children, grandchildren, etc.) under this one.
+                A parent is visited before any children.
+                An action (function) is run for each, which can return false to stop iteration:
+
+                    foreach_cm_node( cm_node::root(),
+                        [&]( cm_node device, size_t depth ) -> bool
+                        {
+                            ...
+                        } );
+
+                Returns whether iteration was stopped. False if the whole tree was finished.
+            */
+            template< class F >
+            bool foreach( F fn, size_t depth = 1 )
```

Comment:
This name has been previously reported as problematic, specifically with QTCreator
https://github.com/IntelRealSense/librealsense/pull/2470